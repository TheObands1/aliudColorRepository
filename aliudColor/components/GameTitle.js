import React from "react";
import { Text, View, StyleSheet } from "react-native";

const GameTitle = () => {
    return(
        <View style={{ flexDirection: "row" }}>
            <Text style={[TitleStyle.fontStyle, { color: "#ecf0f1" }]}>aliud</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#E64C3C" }, { textTransform: 'uppercase'}]}>c</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#E57E31" }]}>o</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#F1C431" }]}>l</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#68CC73" }]}>o</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#3998DB" }]}>r</Text>
        </View>
    );
}

const TitleStyle = StyleSheet.create({
    fontStyle: {
      fontSize: 50,
      fontFamily: "dogbyte"
    }
   });

export { GameTitle };