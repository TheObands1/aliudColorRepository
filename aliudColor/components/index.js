export * from './GameTitle'
export * from "./GameplayComponents/GameGrid"
export * from "./GameplayComponents/LoseView"
export * from "./GameplayComponents/pauseComponents/pauseView"
export * from "./GameplayComponents/bottomComponents/BottomArea"