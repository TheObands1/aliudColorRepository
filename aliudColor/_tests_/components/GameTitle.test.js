import React from 'react';
import renderer from 'react-test-renderer';
import { GameTitle } from "../../components";

describe('tests of the GameTitle', () => {
    it('Create the GameTitle and expect it to have 6 children, which represent the six letters of "aliudColor"', () => {
      const tree = renderer.create(<GameTitle />).toJSON();
      expect(tree.children.length).toBe(6);
    });

    it('Render the component correctly', () => {
        const tree = renderer.create(<GameTitle />).toJSON();
        expect(tree).toMatchSnapshot();
      });

  });