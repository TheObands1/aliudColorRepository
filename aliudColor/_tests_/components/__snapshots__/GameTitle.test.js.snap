// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests of the GameTitle Render the component correctly 1`] = `
<View
  accessibilityLabel="aliudColor"
  accessible={true}
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "dogbyte",
        },
        Object {
          "color": "#ecf0f1",
          "fontSize": 55,
        },
      ]
    }
  >
    aliud
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "dogbyte",
        },
        Object {
          "color": "#E64C3C",
          "fontSize": 55,
          "textTransform": "uppercase",
        },
      ]
    }
  >
    c
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "dogbyte",
        },
        Object {
          "color": "#E57E31",
          "fontSize": 55,
        },
      ]
    }
  >
    o
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "dogbyte",
        },
        Object {
          "color": "#F1C431",
          "fontSize": 55,
        },
      ]
    }
  >
    l
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "dogbyte",
        },
        Object {
          "color": "#68CC73",
          "fontSize": 55,
        },
      ]
    }
  >
    o
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "dogbyte",
        },
        Object {
          "color": "#3998DB",
          "fontSize": 55,
        },
      ]
    }
  >
    r
  </Text>
</View>
`;
